compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -147,42 +148,21 @@ pub unsafe fn create_module<'ll>(
     }
 
     // Ensure the data-layout values hardcoded remain the defaults.
-    if sess.target.is_builtin {
-        // tm is disposed by its drop impl
+    {
         let tm = crate::back::write::create_informational_target_machine(tcx.sess);
         llvm::LLVMRustSetDataLayoutFromTargetMachine(llmod, &tm);
 
         let llvm_data_layout = llvm::LLVMGetDataLayoutStr(llmod);
         let llvm_data_layout = str::from_utf8(CStr::from_ptr(llvm_data_layout).to_bytes())
             .expect("got a non-UTF8 data-layout from LLVM");
 
-        // Unfortunately LLVM target specs change over time, and right now we
-        // don't have proper support to work with any more than one
-        // `data_layout` than the one that is in the rust-lang/rust repo. If
-        // this compiler is configured against a custom LLVM, we may have a
-        // differing data layout, even though we should update our own to use
-        // that one.
```

Comment:
I assume the effects of this actually happening range from compiler ICEs and crashes to silent UB at the worst? 

I'm not sure I understand why this was disabled if you used a custom LLVM. As far as I can tell, all the APIs we're calling should work with a custom LLVM as well. Is there some other potential issue I'm missing? 🙂

Reply:
I get the impression that it was disabled for custom LLVM not because we expected APIs to be missing but just because we thought that this might be an undesirable issue to hit for custom LLVM users - but I can't really work out why mismatched data layouts would happen for custom LLVM users and simultaneously not be an issue that needs resolved.

Reply:
Yeah, agreed! It might even be *more* useful for custom LLVM users if they're trying to light up support for a new target or something.